python/ray/workflow/step_function.py

```diff
@@ -132,11 +122,15 @@ def options(
         Returns:
             The step function itself.
         """
-        # TODO(suquark): The options seems drops items that we did not
-        # specify (e.g., the name become "None" if we did not pass
-        # name to the options). This does not seem correct to me.
+        name = name or self._name
+        validate_user_metadata(metadata)
+        if metadata is not None:
+            metadata = {**self._user_metadata, **metadata}
+        else:
+            metadata = self._user_metadata
         step_options = WorkflowStepRuntimeOptions.make(
             step_type=StepType.FUNCTION,
+            existing_options=self._step_options,
```

Comment:
I got your point here, but maybe let's don't pass existing_options to WorkflowStepRuntimeOptions since this seems like pure application logic.

What if the StepFunction itself has field call options and this function update the field and constructor a new WorkflowStepFunction?